referencesrv/parser/pub.py

```diff
@@ -493,11 +538,17 @@ def publisher_idx(self, ref_words):
         :param ref_words:
         :return:
         """
-        for i, tokens in enumerate(ref_words):
-            match = self.academic_publishers_and_locations.search(tokens)
-            if match:
-                if match.group() == tokens:
+        try:
+            for i, tokens in enumerate(ref_words):
+                match = self.academic_publishers_and_locations.findall(tokens)
+                # concatenate all the matched words, then split into words
+                match = ' '.join([' '.join([word for word in list(filter(None, m))]) for m in match]).split()
+                # then compare all the words, make sure all the words in tokens were matched
+                token_words = self.WORDS.findall(tokens)
+                if len(set(match) & set(token_words)) >= len(token_words):
                     return i
+        except:
```

Comment:
i'm not sure why the catch block is needed - but I'd note that the following code (if it has bugs) will be forever silent; please add comment on the line 551 (something to the effect: yes, it is swallowing all exceptions and that is okay) or consider making the try/catch block ignore only certain exceptions

Reply:
I actually have comments on the line that I am calling this function
https://github.com/golnazads/reference_service/blob/master/referencesrv/parser/pub.py#L187
for completeness I shall add the same comment to except block